### ISSUE TITLE
False positive for SA1122 inside switch expression 

### ISSUE THREAD

**Issue by sungam3r** [DotNetAnalyzers/StyleCopAnalyzers#3124]
```c#
 internal class Test
    {
        internal static string Echo(string source)
            => source switch
            {
                "" => "empty", // SA1122  Use string.Empty for empty strings
                _ => source
            };
    }
```
After applying "Replace with string.Empty" quick fix:

```c#
 internal class Test
    {
        internal static string Echo(string source)
            => source switch
            {
                string.Empty => "empty", // CS0150  A constant value is expected
                _ => source
            };
    }
```

**Comment by sharwell** [comment]
Duplicate of #3028
